job/server/src/main/java/alluxio/job/plan/migrate/MigrateDefinition.java

```diff
@@ -197,7 +200,26 @@ private void checkMigrateValid(MigrateConfig config, FileSystem fs) throws Excep
     Set<Pair<WorkerInfo, ArrayList<MigrateCommand>>> result = Sets.newHashSet();
 
     for (Map.Entry<WorkerInfo, ArrayList<MigrateCommand>> assignment : assignments.entrySet()) {
-      result.add(new Pair<>(assignment.getKey(), assignment.getValue()));
+
+      // split the list of MigrateCommands for a given worker into JOBS_PER_WORKER.
+
+      ArrayList<MigrateCommand> migrateCommands = assignment.getValue();
+
+      ArrayList<ArrayList<MigrateCommand>> splittedCommands = Lists.newArrayList();
+
+      for (int i = 0; i < JOBS_PER_WORKER; i++) {
+        splittedCommands.add(Lists.newArrayList());
+      }
+
+      for (int i = 0; i < migrateCommands.size(); i++) {
+        splittedCommands.get(i % JOBS_PER_WORKER).add(migrateCommands.get(i));
+      }
```

Comment:
Do you mind removing some of the white space? Also, would `Lists.partition` work here?

Reply:
Oh I think Lists.partition is probably great.